source/atomics.tex

```diff
@@ -928,14 +931,54 @@
 \end{note}
 
 \pnum
-\begin{note} The \tcode{memcpy} and \tcode{memcmp} semantics of the compare-and-exchange
-operations may result in failed comparisons for values that compare equal with
-\tcode{operator==} if the underlying type has padding bits, trap bits, or alternate
+\begin{note}
+Under cases where \tcode{memcpy} and \tcode{memcmp} semantics of the compare-and-exchange
+operations apply, the outcome might be failed comparisons for values that compare equal with
+\tcode{operator==} if the underlying type has trap bits or alternate
 representations of the same value. Notably, on implementations conforming to
 ISO/IEC/IEEE 60559, floating-point \tcode{-0.0} and \tcode{+0.0}
 will not compare equal with \tcode{memcmp} but will compare equal with \tcode{operator==},
 and NaNs with the same payload will compare equal with \tcode{memcmp} but will not
-compare equal with \tcode{operator==}.\end{note}
+compare equal with \tcode{operator==}.
+\end{note}
+\begin{note}
+Because compare-and-exchange acts on an object's value representation,
+padding bits that never participate in the object's value representation
+are ignored. As a consequence, the following code is guaranteed to avoid
+spurious failure:
+\begin{codeblock}
+struct padded {
+  char clank = 0x42;
+  // Padding here.
+  unsigned biff = 0xC0DEFEFE;
+};
+atomic<padded> pad = ATOMIC_VAR_INIT({});
```

Comment:
Leaving as is then? We can change it later if that's appropriate.

Reply:
Yes, I think you should leave it as-is and editorially remove uses of ATOMIC_VAR_INIT when P0833R1 is voted in. This was discussed in CWG and people decided to keep ATOMIC_VAR_INIT in my paper for now.

Reply:
We don't need P0833R1. That permits

    atomic<padded> pad = {};

but either of the forms I suggested work fine without P0833, because they use the atomic<T>(T) constructor, not the default constructor.